adapters/sharethrough/sharethrough.go

```diff
@@ -34,9 +34,10 @@ type SharethroughAdapter struct {
 
 func (a SharethroughAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters.RequestData, []error) {
 	var reqs []*adapters.RequestData
+	var domain = Util{}.parseDomain(request.Site.Page)
```

Comment:
I see the `TestSuccessMakeRequests` inside `adapters/sharethrough/sharethrough_test.go` was modified to work with `openrtb.Site` instead of `openrtb.App`. Even under this new conditions, are we sure we don't want to add a NULL check on `request.Site` here just to make sure we don't ever get a segmentation fault? If we comment out the `openrtb.BidRequest.Site` field declaration in `TestSuccessMakeRequests`  like this:
```
 43 func TestSuccessMakeRequests(t *testing.T) {
 44     stubReq := &adapters.RequestData{
 45 +-- 12 lines: Method: "POST",-----------------------------------------------------
 57     }{
 58         "Generates expected Request": {
 59             input: &openrtb.BidRequest{
 60                 //Site: &openrtb.Site{
 61                 //  Page: "test.com",
 62                 //},
 63                 Device: &openrtb.Device{
 64                     UA: "Android Chome/60",
 65 +-- 20 lines: },------------------------------------------------------------------
 85     for testName, test := range tests {
 86         t.Logf("Test case: %s\n", testName)
 87
 88         output, actualErrors := adapter.MakeRequests(test.input)
adapters/sharethrough/sharethrough_test.go
```
We get the a segmentation fault exactly here (`adapters/sharethrough/sharethrough.go` line 37). Is this a realistic scenario? Complete message:
```
╰─➤  go test
--- FAIL: TestSuccessMakeRequests (0.00s)
    sharethrough_test.go:86: Test case: Generates expected Request
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x80 pc=0x1474d7e]

goroutine 37 [running]:
testing.tRunner.func1(0xc00017d600)
        /usr/local/opt/go/libexec/src/testing/testing.go:792 +0x387
panic(0x15168a0, 0x19e6980)
        /usr/local/opt/go/libexec/src/runtime/panic.go:513 +0x1b9
github.com/prebid/prebid-server/adapters/sharethrough.SharethroughAdapter.MakeRequests(0x16454e0, 0xc0003de660, 0xc00037a8c0, 0xc000065d48, 0x1, 0x1, 0xc0003ea498, 0x174ff30, 0x19b384b)
        /Users/gcarreongutierrez/go/src/github.com/prebid/prebid-server/adapters/sharethrough/sharethrough.go:37 +0x3e
github.com/prebid/prebid-server/adapters/sharethrough.TestSuccessMakeRequests(0xc00017d600)
        /Users/gcarreongutierrez/go/src/github.com/prebid/prebid-server/adapters/sharethrough/sharethrough_test.go:88 +0x6ea
testing.tRunner(0xc00017d600, 0x15d3468)
        /usr/local/opt/go/libexec/src/testing/testing.go:827 +0xbf
created by testing.(*T).Run
        /usr/local/opt/go/libexec/src/testing/testing.go:878 +0x35c
exit status 2
FAIL    github.com/prebid/prebid-server/adapters/sharethrough   0.026s
```

Reply:
@guscarreon The removal of `openrtb.App` is actually not related to the addition of `openrtb.Site`, I just cleaned up the test while I was on it. All usages of `openrtb.App` had been removed in the previous PR, see this commit: https://github.com/sharethrough/prebid-server/commit/74b2f721867ed8871a8168a7e6ca057275a6d9ba

Regarding adding a null check on `request.Site`, as far as I understand it, Prebid core will always make the value accessible, feel free to correct me but I don't think a null `openrtb.Site` is a realistic scenario.

Reply:
@maphe, prebid-server core will throw an error if both `openrtb.Site` and `openrtb.App` are NULL or not NULL. In other words something similar to:
```
if (req.Site == nil && req.App == nil) || (req.Site != nil && req.App != nil) {
    //throw ERROR
    return
} 
```
Therefore, we might risk a Segmentation Fault in processing a given request coming with an `openrtb.Site` object that is NULL and an `openrtb.App` that is not. Do you think it's worth adding this NULL check in your adapter to cover up for this scenario? I realize you are not using `openrtb.App` anymore but, according to how prebid-server core handles things, `openrtb.Site` might actually be NULL.

Reply:
@guscarreon, got it! thanks for the details. We don't support in app placements so in theory we shouldn't have any of those, but still I added a null check on the `request.Site` for extra safety.